File: Java/MonolithicApp/src/main/java/com/example/route/dto/ConnectionUsageDto.java
package com.example.route.dto;

public class ConnectionUsageDto {
    private String connectionId;
    private String from;
    private String to;
    private long usageCount;

    public ConnectionUsageDto(String connectionId, String from, String to, long usageCount) {
        this.connectionId = connectionId;
        this.from = from;
        this.to = to;
        this.usageCount = usageCount;
    }

    public String getConnectionId() { return connectionId; }
    public String getFrom() { return from; }
    public String getTo() { return to; }
    public long getUsageCount() { return usageCount; }
}

File: NET/MicroserviceApp/RouteService/DTOs/ConnectionUsageDto.cs
namespace RouteService.DTOs
{
    public class ConnectionUsageDto
    {
        public string ConnectionId { get; set; }
        public string From { get; set; }
        public string To { get; set; }
        public long UsageCount { get; set; }

        public ConnectionUsageDto(string connectionId, string from, string to, long usageCount)
        {
            ConnectionId = connectionId;
            From = from;
            To = to;
            UsageCount = usageCount;
        }
    }
}
